src/Command/GenerateCreditMemo.php

```diff
@@ -13,34 +13,35 @@
 
 namespace Sylius\RefundPlugin\Command;
 
-use Sylius\RefundPlugin\Model\OrderItemUnitRefund;
 use Sylius\RefundPlugin\Model\ShipmentRefund;
+use Sylius\RefundPlugin\Model\UnitRefundInterface;
 use Webmozart\Assert\Assert;
 
 class GenerateCreditMemo
 {
-    private string $orderNumber;
+    private array $shipments = [];
 
-    private int $total;
+    public function __construct(
+        private string $orderNumber,
+        private int $total,
+        /** @var array|UnitRefundInterface[] */
+        private array $units,
+        private string|array $comment,
+    ) {
+        $args = func_get_args();
```

Comment:
What about something like that?
```suggestion
    public function __construct(
        private string $orderNumber,
        private int $total,
        /** @var array|UnitRefundInterface[] */
        private array $units,
        private string|array $comment,
        string|array $shipmentsOrComment,
        ?string $comment = null,
    ) {
        if (is_array($shipmentsOrComment)) { // trigger deprecation }
        if (null !== $comment)) { // trigger deprecation }
        if (null !== $comment && is_string($shipmentsOrComment)) { // trigger deprecation }
```
I'm still considering how much we should support named arguments tho... but this could be handled with error handling and adding null to `shipmentsOrComment`

Reply:
How should it work? I see 4th and 6th arguments have the same name. Another thing is that In the previous implementation, there were 5 args and now we introduce another but the last one should never be used 🤔